src/guidellm/data/preprocessors/formatters.py

```diff
@@ -191,27 +197,25 @@ def __call__(  # noqa: C901, PLR0912, PLR0915
             if not prefix:
                 continue
 
-            stats = text_stats(prefix)
-            if (num_chars := stats.get("num_chars")) is not None:
-                input_metrics.text_characters = (
-                    input_metrics.text_characters or 0
-                ) + num_chars
-            if (num_words := stats.get("num_words")) is not None:
-                input_metrics.text_words = (input_metrics.text_words or 0) + num_words
+            input_metrics.text_characters = (
```

Comment:
I would like to keep the counting logic unified in the text_stats function. We could update it so that it pushes to a UsageMetrics instance directly / make that a functoin of the UsageMetrics class, but this code is duplicated and if we add anymore stats, it will make it hard to keep updated

Reply:
Fixed. I think the new code is a much more elegant solution.